### ISSUE TITLE
One menu always active

### ISSUE THREAD

**Issue by ingeva** [joomla/joomla-cms#6341]
#### Steps to reproduce the issue

Check this screenshot
http://norskveteranvognklubb.no/Screenshot.png
or
http://veteranvogn.no

#### Expected result

Can't reproduce

#### Actual result

as show in screenshot

#### System information (as much as possible)

Joomla! 3.4. 

#### Additional comments

The third menu always has one active menu. Menus 1 and 2 are shown correctly. The background for class="active" and class="current" should be evident when viewing the screenshot or accessing the page. The html code shows class="active" for the darker light-blue menu items. This is probably a bug in the Joomla! code, and I hoped it would be fixed in 3.4, but it's still there.
I use three menus because I want a header to be shown for number 2 and 3. I don't know what will happen with more than 3 menus.

Category set to Code style. Couldn't find anything for "menu".

**Comment by bertmert** [comment]
Cannot reproduce.
Did you already:
- deactivate cache (Joomla and in module itself).
- delete Joomla cache
- backend > Menus > Menu Manager: click rebuild button

**Comment by ingeva** [comment]
I've done that now, and there's no change. The same error occurs with the site on the remote server, and on my own, local one. The error is in the HTML code, as can be seen by viewing the source.

**Comment by Hackwar** [comment]
Looking at the site right now, the described error is not visible. Can this be closed?

**Comment by ingeva** [comment]
What about the html code generated? It shows "acive" for an element that is NOT active!

<li class="item-152"><a href="/tekniske-tips" >Tekniske tips</a></li><li class="item-140 active"><a href="/eldre-artikler" >Eldre artikler</a></li></ul>

**Comment by bertmert** [comment]
"described error is not visible"
It is (menu item "Eldre artikler" has class active on all pages). But I couldn't reproduce it anyhow.
@ingeva: 
Is there an override in your template located at /templates/beez3/html/mod_menu/? Deactivate!
Try with template protostar and also with unchanged original beez3.
Create new menu module with standard settings. Deactivate old one

**Comment by ingeva** [comment]
The menu module has not been touched. Only one CSS file has been added: image.css.
HTML with protostar is unchanged:
<ul class="nav menu">
<li class="item-152"><a href="/tekniske-tips" >Tekniske tips</a></li><li class="item-140 active"><a href="/eldre-artikler" >Eldre artikler</a></li></ul>

So: Still active, although the CSS of "active" doesn't show it.

**Comment by ingeva** [comment]
I forgot the unchanged beez3. Of course, it's the same:
<ul class="nav menu">
<li class="item-152"><a href="/tekniske-tips" >Tekniske tips</a></li><li class="item-140 active"><a href="/eldre-artikler" >Eldre artikler</a></li></ul>

Again, nothing has changed. The only difference is that in my CSS, "active" has another background color. The HTML is unchanged. This bug is not in the CSS. The CSS only reveals it.

**Comment by ingeva** [comment]
Instead of being satisfied with the upgrade to 3.4.0, I made a complete replacement of all the files from the previous package, after deleting every file and directory except the one containing out own addition. I also removed our additional image.css file.

There was not change.
I believe the error lies in the file /modules/mod_menu/helper.php, but my php skill is far from what is required to investigate further.
Of course, there's a way to camouflage the error: I could ignore the "active" class, but I don't want to do that. It's useful.

**Comment by Harmageddon** [comment]
What type of menu item is "Eldre artikel"? <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6341">issues.joomla.org/joomla-cms/6341</a>.</sub>

**Comment by losedk** [comment]
Is the error still present?

**Comment by ingeva** [comment]
Harmageddon: Type of menu is category blog for "tekniske tips", and category list for "Eldre artikler".
For testing I changed to single article for both of them, but no change.

losedk: There is still no change. Next step will be to add a third menu to another Joomla! site, so I can see if the same code can give a different result. If so, we have a mystery on our hands. :)
However, that will have to wait after I have completed other more important tasks.

**Comment by losedk** [comment]
Then I don't totally understand the problem. The li has an active call as it should. Could you point to the issue in the HTML output, which you mention?

**Comment by ingeva** [comment]
It should have an active class when it it is active, and if it is not active, it should not. But it does. I have even removed the active class from the cached mod_menu file - to no avail.

**Comment by losedk** [comment]
Is it LMK-Nytt?

![skaermbillede 2015-03-12 kl 11 04 52](https://cloud.githubusercontent.com/assets/1738811/6615821/a1d882de-c8a7-11e4-809b-98a4978e557f.png)

**Comment by losedk** [comment]
Ahh I see the problem now

**Comment by brianteeman** [comment]
I am glad you do because I dont.

The first item is the parent
the sub item is the child
When the child is active then so is the parent
That is the correct behaviour
If you look you will see that the first item has classes of active AND
parent
the sub item just has classes of active

For me this is the expected and correct behaviour. This is how menu
highlighting works

Example
Sport
--football
--cricket
--athletics

If athletics is the current url  then the correct classes that are applied
to the menu are
Sport (parent active)
--football
--cricket
--athletics (active)

**Comment by Hackwar** [comment]
The issue is, that no matter on which page you are, "Eldre artikler" has a class of "active", even on the homepage.

I still can't replicate that behavior. Did you click on "rebuild" in the menu manager? Maybe that fixes the issue.

**Comment by brianteeman** [comment]
OK so I misunderstood the query - thanks for clarifying Hannes

I am 99% certain that this is a simple config issue but without admin
access I cant tell for certain

On 12 March 2015 at 10:41, Hannes Papenberg <notifications@github.com>
wrote:

> The issue is, that no matter on which page you are, "Eldre artikler" has a
> class of "active", even on the homepage.
>
> I still can't replicate that behavior. Did you click on "rebuild" in the
> menu manager? Maybe that fixes the issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/6341#issuecomment-78457134>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by ingeva** [comment]
I have done everything you suggested so far. The "Eldre artikler" menu always seems to be active even if it is not - and neither is the parent. I can select any of the other menus. This menu is marked as active nonetheless.
Please re-read the subject of the topic.

**Comment by brianteeman** [comment]
As no one can replicate this issue and we are not mind readers it is
impossible to debug this any further without access to the site. If we
could replicate the issue it would be easy to debug but we cant so
providing access is the only option

On 12 March 2015 at 12:03, ingeva <notifications@github.com> wrote:

> I have done everything you suggested so far. The "Eldre artikler" menu
> always seems to be active even if it is not - and neither is the parent. I
> can select any of the other menus. This menu is marked as active
> nonetheless.
> Please re-read the subject of the topic.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/6341#issuecomment-78467522>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by ingeva** [comment]
I understand, but I'm reluctant go provide access to an active site, which is mainly in Norwegian so it might not be so easy to understand for non-Norwegians.
However, I just found a way to overcome to problem. I created a new menu item, identical (or so I thought) to the "always active" one. It did not have the problem. I then deleted the "Eldre artikler" item and renamed the new one. Problem gone!
(Eldre artikler = Older articles) - Norwegian is easy for those who know it! :)
This was a tricky one. I may have unwillingly set some parameter that caused the problem. I just wish I knew which one, but when the problem is gone I don't wish to spend more time on it. Thanks for all your help - after all, this was nothing serious. None of the users have even commented it, but without your input I would probably still be without a clue.
